### PR TITLE
ECOM-4799 Updating documentation to reflect programs service prereqs …

### PR DESCRIPTION
**Pull request by mikedikan**
…and changing port number to avoid collision with documented ecommerce default

### CODE REVIEW

**Comment by mikedikan** [comment]
ECOM-4799 Work is complete.  Waiting for thumbs from @rlucioni and/or @schenedx

File: docs/prerequisites.rst [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    $ nvm use
+
+If you already have Node.js version 4.4.6 Installed, NVM will switch to using it.  If not, use this command to install
+(and switch to) it:
+
+.. code-block:: bash
+
+    $nvm install
+
+Virtualenvwrapper/pip Installation
```
Comment (schenedx) [inline docs/prerequisites.rst]: Is this also optional?
I know this is not optional for standalone. What about devstack environment?
If this is not optional, can you move this section above the optional NVM section?

**Comment by schenedx** [comment]
Besides the lone comment, rest LGTM

File: docs/prerequisites.rst (hunk 1 above)
Comment (mikedikan) [inline docs/prerequisites.rst]: As far as I know, the virtualenvwrapper installation is also _needed_ for devstack but is already available.  That said, the documents make the assumption (which was the case before) that you will be setting up the program service locally instead of on devstack.  I'm open to wording changes, but they would need to be applied throughout the two (prereq and getting_started) documents uniformly.

File: docs/getting_started.rst [hunk 2]
```diff
@@ -23,6 +23,14 @@ If you need such overrides, create a file :file:`programs/settings/private.py`.
 read by :file:`programs/settings/local.py`, but ignored by Git. 
 For an example, see `programs/settings/private.py.example`.
 
+LMS Admin
+---------
+
+Continuing configuration for the programs service integration will require access to the LMS admin portal for Django.
+If you don't already have access, please follow the instructions at the following link:
+
+.. _LMS Admin Access: https://openedx.atlassian.net/wiki/pages/viewpage.action?pageId=94306332
```
Comment (rlucioni) [inline docs/getting_started.rst]: This link isn't being rendered. Here's an example of how to render it inline: https://github.com/edx/programs/pull/142/files#diff-9bc8bf6e8d9db020cefce1fbc0426795R4.

File: docs/getting_started.rst [hunk 3]
```diff
@@ -113,6 +124,43 @@ OIDC client via LMS admin.
 
 .. code-block:: bash
 
-    $ python manage.py runserver 8003
+    $ python manage.py runserver 8004
```
Comment (rlucioni) [inline docs/getting_started.rst]: Alternatively, run `make serve`.

File: docs/getting_started.rst [hunk 4]
```diff
@@ -113,6 +124,43 @@ OIDC client via LMS admin.
...
+
+.. code-block:: bash
+
+    $ vagrant ssh -- -R 8004:127.0.0.1:8004
+
+Confirm API Access
+------------------
+The server can be acessed directly by navigating to the browseable api docroot:
+
+.. _Programs API: http://localhost:8004/api/v1/
```
Comment (rlucioni) [inline docs/getting_started.rst]: This link is also lost when this document is rendered. See above for an example of how to include it inline.

File: docs/getting_started.rst [hunk 5]
```diff
@@ -113,6 +124,43 @@ OIDC client via LMS admin.
...
+VM, the LMS needs a tunnel back out to the host machine to hit the service on its listening port.  Using the port you
+configured above for your programs server, open a reverse tunnel with the following command:
+
+.. code-block:: bash
+
+    $ vagrant ssh -- -R 8004:127.0.0.1:8004
+
+Confirm API Access
+------------------
+The server can be acessed directly by navigating to the browseable api docroot:
```
Comment (rlucioni) [inline docs/getting_started.rst]: > Verify your setup by navigating to the browsable API.

File: docs/getting_started.rst [hunk 6]
```diff
@@ -113,6 +124,43 @@ OIDC client via LMS admin.
...
+
+.. _Programs API: http://localhost:8004/api/v1/
+
+Confirm that you can see an HTML REST interface, but one that informs you that you need to authenticate to continue.
+
+By clicking the 'Log In' link in the top right of the page, we can confirm that the OAUTH was configured properly.  If
```
Comment (rlucioni) [inline docs/getting_started.rst]: > ...we can confirm that OIDC was configured properly.

File: docs/getting_started.rst [hunk 7]
```diff
@@ -113,6 +124,43 @@ OIDC client via LMS admin.
...
+----------------
+
+Once you have access, you will be able to browse the API.  At the top level, you will see (as of this writing) 3 red
+links - one for programs, one for course codes, and one for organizations.  Clicking any of these will take you to the
+listing page for that object type.  If viewing the listing results for programs (the fixture data should have some data
+across all of the object types) you can append the ID of an entry in the results to inspect the details of a single
+entry.
+
+The REST UI will also provide a form to allow you to modify the program record and issue either a PUT or PATCH HTTP
+Request to do so.
```
Comment (rlucioni) [inline docs/getting_started.rst]: s/Request/request

File: docs/prerequisites.rst [hunk 8]
```diff
@@ -0,0 +1,91 @@
+Software Prerequisites
+======================
+
+- programs edx code (github repository)
+.. _programs: https://github.com/edx/programs
```
Comment (rlucioni) [inline docs/prerequisites.rst]: These links are also all missing from the rendered document.

File: docs/prerequisites.rst [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+
+- Node.js 4.4.6 (download or use NVM - explained below)
+.. _Node 4.4.6: https://nodejs.org/download/release/v4.4.6/
+
+- NPM 2.15.5 (Inlcuded with Node.js download)
```
Comment (rlucioni) [inline docs/prerequisites.rst]: s/NPM/npm

File: docs/prerequisites.rst [hunk 10]
```diff
@@ -0,0 +1,91 @@
+Software Prerequisites
+======================
+
+- programs edx code (github repository)
```
Comment (rlucioni) [inline docs/prerequisites.rst]: > Clone of [edx/programs](https://github.com/edx/programs).

File: docs/prerequisites.rst [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+.. _programs: https://github.com/edx/programs
+
+- Node.js 4.4.6 (download or use NVM - explained below)
+.. _Node 4.4.6: https://nodejs.org/download/release/v4.4.6/
+
+- NPM 2.15.5 (Inlcuded with Node.js download)
+.. _Node 4.4.6: https://nodejs.org/download/release/v4.4.6/
+
+- pip 8.1.2
```
Comment (rlucioni) [inline docs/prerequisites.rst]: If you're going to list pip, you should also list the supported Python versions. At the moment, 2.7.x.

File: docs/prerequisites.rst [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+.. Follow instructions below
+
+- Virtualenvwrapper (follow linked instructions)
```
Comment (rlucioni) [inline docs/prerequisites.rst]: This should be listed as a pre-requisite. The instructions below walk the user through installation.

File: docs/prerequisites.rst [hunk 13]
```diff
@@ -0,0 +1,91 @@
...
+.. _programs: https://github.com/edx/programs
+
+- Node.js 4.4.6 (download or use NVM - explained below)
+.. _Node 4.4.6: https://nodejs.org/download/release/v4.4.6/
+
+- NPM 2.15.5 (Inlcuded with Node.js download)
+.. _Node 4.4.6: https://nodejs.org/download/release/v4.4.6/
+
+- pip 8.1.2
+.. Follow instructions below
```
Comment (rlucioni) [inline docs/prerequisites.rst]: This text is missing from the rendered document. I think you can get rid of it.

File: docs/prerequisites.rst [hunk 14]
```diff
@@ -0,0 +1,91 @@
...
+
+- Virtualenvwrapper (follow linked instructions)
+.. Follow instructions below
+
+
+NVM (Optional)
```
Comment (rlucioni) [inline docs/prerequisites.rst]: s/NVM/nvm

File: docs/prerequisites.rst [hunk 15]
```diff
@@ -0,0 +1,91 @@
...
+.. code-block:: bash
+
+    $ nvm use
+
+If you already have Node.js version 4.4.6 Installed, NVM will switch to using it.  If not, use this command to install
+(and switch to) it:
+
+.. code-block:: bash
+
+    $nvm install
```
Comment (rlucioni) [inline docs/prerequisites.rst]: Nitpick: missing whitespace between $ and nvm.

File: docs/prerequisites.rst [hunk 16]
```diff
@@ -0,0 +1,91 @@
...
+
+With NVM, you can easily switch from one node version to another, and even declare the version of node that
+a project needs to work with in a hidden file that is read in by nvm in the working directory. If using NVM,
+simply change directories to the programs directory from the github checkout and use this command:
+
+.. code-block:: bash
+
+    $ nvm use
+
+If you already have Node.js version 4.4.6 Installed, NVM will switch to using it.  If not, use this command to install
```
Comment (rlucioni) [inline docs/prerequisites.rst]: s/Installed/installed

File: docs/prerequisites.rst (hunk 9 above)
Comment (rlucioni) [inline docs/prerequisites.rst]: s/Inlcuded/included

File: docs/prerequisites.rst [hunk 17]
```diff
@@ -0,0 +1,91 @@
...
+(and switch to) it:
+
+.. code-block:: bash
+
+    $nvm install
+
+Virtualenvwrapper/pip Installation
+----------------------------------
+
+1. On your local machine, if you haven't already, install pip.
```
Comment (rlucioni) [inline docs/prerequisites.rst]: It may be worth linking to http://docs.python-guide.org/en/latest/dev/virtualenvs/#virtualenvwrapper.

File: docs/prerequisites.rst [hunk 18]
```diff
@@ -0,0 +1,91 @@
+Software Prerequisites
```
Comment (rlucioni) [inline docs/prerequisites.rst]: This document will not appear in the rendered docs unless you list it [here](https://github.com/edx/programs/blob/master/docs/index.rst).

File: docs/prerequisites.rst [hunk 19]
```diff
@@ -0,0 +1,91 @@
...
+-----------------------
+
+7. Confirm that you have the lms running in devstack.  If you don't, please consult the wiki document for instructions.
+
+.. _Running Devstack: https://openedx.atlassian.net/wiki/display/OpenOPS/Running+Devstack
+
+8. Refer to the programs getting_started document for setting up the programs service now that the prereqs
+(node, pip, virtualenv) are met.
+
+.. _getting_started: http://open-edx-programs-guide.readthedocs.io/en/latest/getting_started.html
```
Comment (rlucioni) [inline docs/prerequisites.rst]: This should be a relative link to the getting started page.

File: programs/settings/private.py.example [hunk 20]
```diff
@@ -1,5 +1,5 @@
-SOCIAL_AUTH_EDX_OIDC_KEY = 'replace-me'
-SOCIAL_AUTH_EDX_OIDC_SECRET = 'replace-me'
-SOCIAL_AUTH_EDX_OIDC_URL_ROOT = 'http://127.0.0.1:8000/oauth'
+SOCIAL_AUTH_EDX_OIDC_KEY = 'programs-key'
+SOCIAL_AUTH_EDX_OIDC_SECRET = 'programs-secret'
+SOCIAL_AUTH_EDX_OIDC_URL_ROOT = 'http://127.0.0.1:8000/oauth2'
```
Comment (rlucioni) [inline programs/settings/private.py.example]: This shouldn't be necessary, it's already configured in [local settings](https://github.com/edx/programs/blob/master/programs/settings/local.py#L54).

**Comment by rlucioni** [comment]
@mikedikan I've completed my first pass.

File: docs/getting_started.rst (hunk 2 above)
Comment (mikedikan) [inline docs/getting_started.rst]: fixed in latest changes.

File: docs/getting_started.rst (hunk 3 above)
Comment (mikedikan) [inline docs/getting_started.rst]: Added section about make serve

File: docs/getting_started.rst (hunk 5 above)
Comment (mikedikan) [inline docs/getting_started.rst]: used suggested wording that was more concise and accurate

File: docs/getting_started.rst (hunk 4 above)
Comment (mikedikan) [inline docs/getting_started.rst]: fixed

File: docs/getting_started.rst (hunk 6 above)
Comment (mikedikan) [inline docs/getting_started.rst]: Updated OAUTH to OIDC to be more precise.

File: docs/getting_started.rst (hunk 7 above)
Comment (mikedikan) [inline docs/getting_started.rst]: fixed.

File: docs/prerequisites.rst (hunk 18 above)
Comment (mikedikan) [inline docs/prerequisites.rst]: added

File: docs/prerequisites.rst (hunk 10 above)
Comment (mikedikan) [inline docs/prerequisites.rst]: updated wording.

File: docs/prerequisites.rst (hunk 8 above)
Comment (mikedikan) [inline docs/prerequisites.rst]: fixed.

File: docs/prerequisites.rst (hunk 11 above)
Comment (mikedikan) [inline docs/prerequisites.rst]: added parenthetical mention of python 2.7

File: docs/prerequisites.rst (hunk 13 above)
Comment (mikedikan) [inline docs/prerequisites.rst]: 86'd

File: docs/prerequisites.rst (hunk 14 above)
Comment (mikedikan) [inline docs/prerequisites.rst]: fixed.

File: docs/prerequisites.rst (hunk 16 above)
Comment (mikedikan) [inline docs/prerequisites.rst]: fixed.

File: docs/prerequisites.rst (hunk 17 above)
Comment (mikedikan) [inline docs/prerequisites.rst]: added.

File: docs/prerequisites.rst (hunk 19 above)
Comment (mikedikan) [inline docs/prerequisites.rst]: fixed.

File: programs/settings/private.py.example (hunk 20 above)
Comment (mikedikan) [inline programs/settings/private.py.example]: removed redef of SOCIAL_AUTH_EDX_OIDC_ID_TOKEN_DECRYPTION_KEY

**Comment by mikedikan** [comment]
@rlucioni review comments addressed - please confirm at your earliest convenience.

File: programs/settings/private.py.example [hunk 21]
```diff
@@ -1,5 +1,4 @@
-SOCIAL_AUTH_EDX_OIDC_KEY = 'replace-me'
-SOCIAL_AUTH_EDX_OIDC_SECRET = 'replace-me'
-SOCIAL_AUTH_EDX_OIDC_URL_ROOT = 'http://127.0.0.1:8000/oauth'
-SOCIAL_AUTH_EDX_OIDC_ID_TOKEN_DECRYPTION_KEY = SOCIAL_AUTH_EDX_OIDC_SECRET
+SOCIAL_AUTH_EDX_OIDC_KEY = 'programs-key'
+SOCIAL_AUTH_EDX_OIDC_SECRET = 'programs-secret'
+SOCIAL_AUTH_EDX_OIDC_URL_ROOT = 'http://127.0.0.1:8000/oauth2'
```
Comment (rlucioni) [inline programs/settings/private.py.example]: Please remove this line, it's not necessary.

File: docs/getting_started.rst [hunk 22]
```diff
@@ -23,6 +23,14 @@ If you need such overrides, create a file :file:`programs/settings/private.py`.
 read by :file:`programs/settings/local.py`, but ignored by Git. 
 For an example, see `programs/settings/private.py.example`.
 
+LMS Admin
+---------
+
+Continuing configuration for the programs service integration will require access to the `LMS Admin Portal`_ for Django.
+If you don't already have access, please follow the instructions at the following link:
```
Comment (rlucioni) [inline docs/getting_started.rst]: You mention that access to the LMS admin is required, but there's nothing in these docs about what to do with it. I would move this section to the bottom of this document, retitle as "LMS Integration," and include details about what needs to be done with the LMS admin.

File: docs/getting_started.rst [hunk 23]
```diff
@@ -109,10 +120,57 @@ You can control JavaScript optimization from the project's `build.js` file by mo
 Run the server
 --------------
 The server can be run with `Django's runserver command`_. If you opt to run on a different port, make sure you update
-OIDC client via LMS admin.
+OIDC client via LMS admin.  Also, a Makefile target (make serve) has been defined to provide a shortcut for starting
```
Comment (rlucioni) [inline docs/getting_started.rst]: Since you've got an example below, there's no need for the parenthesized example here.

File: docs/getting_started.rst [hunk 24]
```diff
@@ -109,10 +120,57 @@ You can control JavaScript optimization from the project's `build.js` file by mo
...
+
+Confirm API Access
+------------------
+Verify your setup by navigating to the browsable `Programs API`_:
+
+.. _Programs API: http://localhost:8004/api/v1/
+
+Confirm that you can see an HTML REST interface, but one that informs you that you need to authenticate to continue.
+
+By clicking the 'Log In' link in the top right of the page, we can confirm that the OIDC was configured properly.  If
```
Comment (rlucioni) [inline docs/getting_started.rst]: s/the OIDC/OIDC

File: docs/getting_started.rst [hunk 25]
```diff
@@ -109,10 +120,57 @@ You can control JavaScript optimization from the project's `build.js` file by mo
...
+you already have an active session in your LMS, you should be automatically logged in and redirected to the REST API
+interface.  If not, log in, and you should be redirected back to the browseable API.
```
Comment (rlucioni) [inline docs/getting_started.rst]: s/browseable/browsable

File: docs/prerequisites.rst [hunk 26]
```diff
@@ -0,0 +1,92 @@
...
+.. _edx/programs: https://github.com/edx/programs
+
+- `Node.js 4.4.6`_ (download or use nvm - explained below)
+.. _Node.js 4.4.6: https://nodejs.org/download/release/v4.4.6/
+
+- npm 2.15.5 (included with Node.js download)
+
+- pip 8.1.2 (requires python 2.7.x)
+
+- Virtualenvwrapper (follow linked instructions)
```
Comment (rlucioni) [inline docs/prerequisites.rst]: I think this section can be more concise, especially given what you have below. For example:
```md
- Python 2.7.x (and pip)
- Node.js 4.4.6 (and npm)
- Clone of [edx/programs](https://github.com/edx/programs)
```

File: docs/prerequisites.rst [hunk 27]
```diff
@@ -0,0 +1,92 @@
...
+(and switch to) it:
+
+.. code-block:: bash
+
+    $ nvm install
+
+Virtualenvwrapper/pip Installation
+----------------------------------
+
+For some general background and basic virtualenvwrapper usage, please consult the `python guide on virtualenv`_.
```
Comment (rlucioni) [inline docs/prerequisites.rst]: > please consult the The Hitchhiker's Guide to Python

File: docs/prerequisites.rst [hunk 28]
```diff
@@ -0,0 +1,92 @@
...
+
+    $ sudo easy_install pip
+
+2. Once pip is installed, you will need to install virtualenvwrapper.
+
+.. code-block:: bash
+
+    $ sudo pip install virtualenvwrapper
+
+3. If (like me) you get an error while trying to unstall the 'six' package, modify the command above to ignore six.
```
Comment (rlucioni) [inline docs/prerequisites.rst]: s/unstall/uninstall

File: docs/prerequisites.rst [hunk 29]
```diff
@@ -0,0 +1,92 @@
...
+
+.. code-block:: bash
+
+    $ sudo pip install virtualenvwrapper --ignore-installed six
+
+4. Once `virtualenvwrapper`_ is installed, follow the instructions on setting up virtualenvwrapper:
```
Comment (rlucioni) [inline docs/prerequisites.rst]: Shuffle this sentence around so you don't need the trailing colon at the end and so that it's clear that the link goes to the instructions.

File: docs/prerequisites.rst [hunk 30]
```diff
@@ -0,0 +1,92 @@
...
+.. _virtualenvwrapper: http://virtualenvwrapper.readthedocs.io/en/latest/
+
+5. Create a new virtualenv for programs.
+
+.. code-block:: bash
+
+    $ mkvirtualenv programs
+
+6. You should now see a terminal that has a prompt that looks something like the prompt below.  This signifies that
+your current working directory is the programs directory, and you are currently in the python virtual environment
```
Comment (rlucioni) [inline docs/prerequisites.rst]: s/python/Python

File: docs/prerequisites.rst [hunk 31]
```diff
@@ -0,0 +1,92 @@
...
+called programs.
+
+.. code-block:: bash
+
+    (programs) MikesMacBook:programs mike$
+
+Continuing Installation
+-----------------------
+
+7. Confirm that you have the lms running in devstack.  If you don't, please consult the `Running Devstack`_ wiki document for instructions.
```
Comment (rlucioni) [inline docs/prerequisites.rst]: s/lms/LMS

**Comment by rlucioni** [comment]
@mikedikan I've taken another look.

File: docs/getting_started.rst (hunk 22 above)
Comment (mikedikan) [inline docs/getting_started.rst]: I've updated the language for clarity, and added in a new section (oops) describing the additional steps for api config for LMS->programs communications.

File: docs/getting_started.rst (hunk 23 above)
Comment (mikedikan) [inline docs/getting_started.rst]: removed.

File: docs/getting_started.rst (hunk 24 above)
Comment (mikedikan) [inline docs/getting_started.rst]: fixed.

File: docs/getting_started.rst (hunk 25 above)
Comment (mikedikan) [inline docs/getting_started.rst]: fixed.

File: docs/prerequisites.rst (hunk 26 above)
Comment (mikedikan) [inline docs/prerequisites.rst]: Updated text to something similar to your suggestion

File: docs/prerequisites.rst (hunk 27 above)
Comment (mikedikan) [inline docs/prerequisites.rst]: updated link text to reference actual name of guide

File: docs/prerequisites.rst (hunk 28 above)
Comment (mikedikan) [inline docs/prerequisites.rst]: changed.

File: docs/prerequisites.rst (hunk 29 above)
Comment (mikedikan) [inline docs/prerequisites.rst]: fixed.

File: docs/prerequisites.rst (hunk 30 above)
Comment (mikedikan) [inline docs/prerequisites.rst]: done.

File: docs/prerequisites.rst (hunk 31 above)
Comment (mikedikan) [inline docs/prerequisites.rst]: yup

File: programs/settings/private.py.example (hunk 21 above)
Comment (mikedikan) [inline programs/settings/private.py.example]: 86'd

**Comment by mikedikan** [comment]
@rlucioni Code review comments round 2 should be addressed (I hope)

File: docs/getting_started.rst [hunk 32]
```diff
@@ -23,6 +23,15 @@ If you need such overrides, create a file :file:`programs/settings/private.py`.
 read by :file:`programs/settings/local.py`, but ignored by Git. 
 For an example, see `programs/settings/private.py.example`.
 
+LMS Admin
+---------
+
+Continuing configuration for the programs service integration will require access to the LMS Admin Portal for Django.
```
Comment (rlucioni) [inline docs/getting_started.rst]: s/Admin Portal/LMS Django admin.

File: docs/getting_started.rst [hunk 33]
```diff
@@ -103,16 +115,84 @@ Execute the following command to run the asset pipeline locally:
...
+.. code-block:: bash
+
+    $ make serve
+
+Integration with LMS
+--------------------
+
+In addition to configuring login access using OIDC from the programs service to the LMS application, it is necessary to
+set up the API configuration for requests coming from the LMS to the programs service.  Using your admin credentials,
+Navigate to Programs>Programs api configs.
```
Comment (rlucioni) [inline docs/getting_started.rst]: s/Navigate/navigate

File: docs/prerequisites.rst [hunk 34]
```diff
@@ -0,0 +1,89 @@
+Software Prerequisites
+======================
+
+- Python 2.7.x (and pip)
+.. (Empty comment for new line...)
```
Comment (rlucioni) [inline docs/prerequisites.rst]: Why is this necessary?

File: programs/settings/private.py.example [hunk 35]
```diff
@@ -1,5 +1,3 @@
-SOCIAL_AUTH_EDX_OIDC_KEY = 'replace-me'
-SOCIAL_AUTH_EDX_OIDC_SECRET = 'replace-me'
-SOCIAL_AUTH_EDX_OIDC_URL_ROOT = 'http://127.0.0.1:8000/oauth'
-SOCIAL_AUTH_EDX_OIDC_ID_TOKEN_DECRYPTION_KEY = SOCIAL_AUTH_EDX_OIDC_SECRET
-CORS_ORIGIN_ALLOW_ALL = True
+SOCIAL_AUTH_EDX_OIDC_KEY = 'programs-key'
+SOCIAL_AUTH_EDX_OIDC_SECRET = 'programs-secret'
+SOCIAL_AUTH_EDX_OIDC_URL_ROOT = 'http://127.0.0.1:8000/oauth2'
```
Comment (rlucioni) [inline programs/settings/private.py.example]: All you need to remove from this file is `SOCIAL_AUTH_EDX_OIDC_URL_ROOT`. Everything else should remain.

**Comment by rlucioni** [comment]
@mikedikan I've taken another look, getting close!

File: docs/getting_started.rst (hunk 32 above)
Comment (mikedikan) [inline docs/getting_started.rst]: changed.

File: docs/getting_started.rst (hunk 33 above)
Comment (mikedikan) [inline docs/getting_started.rst]: fixed.

File: docs/prerequisites.rst (hunk 34 above)
Comment (mikedikan) [inline docs/prerequisites.rst]: For consistency in the rendered RTD version, the new line is added by default for the links, and without a link in the first item, it looks funny.

File: programs/settings/private.py.example (hunk 35 above)
Comment (mikedikan) [inline programs/settings/private.py.example]: this was unclear before.  this has been changed.

**Comment by mikedikan** [comment]
@rlucioni please review

File: docs/prerequisites.rst [hunk 36]
```diff
@@ -0,0 +1,89 @@
...
+
+.. code-block:: bash
+
+    $ nvm install
+
+Virtualenvwrapper/pip Installation
+----------------------------------
+
+For some general background and basic virtualenvwrapper usage, please consult the
+Hitchhiker's guide to Python `section on virtualenv`_.
```
Comment (rlucioni) [inline docs/prerequisites.rst]: Nitpick: Hitchhiker's Guide to Python's

File: programs/settings/private.py.example [hunk 37]
```diff
@@ -1,5 +1,3 @@
-SOCIAL_AUTH_EDX_OIDC_KEY = 'replace-me'
-SOCIAL_AUTH_EDX_OIDC_SECRET = 'replace-me'
-SOCIAL_AUTH_EDX_OIDC_URL_ROOT = 'http://127.0.0.1:8000/oauth'
-SOCIAL_AUTH_EDX_OIDC_ID_TOKEN_DECRYPTION_KEY = SOCIAL_AUTH_EDX_OIDC_SECRET
-CORS_ORIGIN_ALLOW_ALL = True
+SOCIAL_AUTH_EDX_OIDC_KEY = 'programs-key'
+SOCIAL_AUTH_EDX_OIDC_SECRET = 'programs-secret'
```
Comment (rlucioni) [inline programs/settings/private.py.example]: Please keep the `SOCIAL_AUTH_EDX_OIDC_ID_TOKEN_DECRYPTION_KEY` line. It's necessary in this example because the secret key is being changed.

**Comment by rlucioni** [comment]
@mikedikan back to you, just a few last comments.

**Comment by mikedikan** [comment]
@rlucioni ready

File: .nvmrc [hunk 38]
```diff
@@ -0,0 +1 @@
+4.4.6
```
Comment (rlucioni) [inline .nvmrc]: The latest LTS appears to be 4.4.7 (https://nodejs.org/en/).

File: docs/prerequisites.rst [hunk 39]
```diff
@@ -0,0 +1,89 @@
+Software Prerequisites
+======================
+
+- Python 2.7.x (and pip)
+.. (Empty comment for new line...)
+- Clone of `edx/programs`_ (github repository)
+.. _edx/programs: https://github.com/edx/programs
+
+- `Node.js 4.4.6`_ (and npm) (download or use nvm - explained below)
```
Comment (rlucioni) [inline docs/prerequisites.rst]: I don't think it makes sense to specify a patch version. 4.4.x should be sufficient.

**Comment by mikedikan** [comment]
@rlucioni please check

**Comment by rlucioni** [comment]
👍 🚢 🎆

**Comment by mikedikan** [comment]
😌
